docs(macros): Improve parameterized macros docs (#1888)
zmkfirmware/zmk: docs/docs/behaviors/macros.md

To prevent issues with longer macros, you can change the size of this queue via the `CONFIG_ZMK_BEHAVIORS_QUEUE_SIZE` setting in your configuration, [typically through your `.conf` file](../config/index.md). For example, `CONFIG_ZMK_BEHAVIORS_QUEUE_SIZE=512` would allow your macro to type about 256 characters.

## Parameterized Macros

Macros can also be "parameterized", allowing them to be bound in your keymap with unique values passed into them, e.g.:

```
    raise_layer {

>         bindings = <&my_parameterized_macro A>

Reviewer: ```suggestion
        bindings = <&my_one_param_macro A>;
```
More consistent with the following section.

Author: Done

    };
```

### Defining Parameterized Macros

> Parameterized macros must be defined using specific values for the `compatible` and `#binding-cells` properties, depending on how many parameters they require (max `2`):

Reviewer: ```suggestion
Parameterized macros must be defined using specific values for the `compatible` and `#binding-cells` properties, depending on how many parameters they require (up to a maximum of two):
```

Author: Done

```dts
/ {
    macros {
        // 0 params macro
        my_macro: my_macro {
            // ...
            compatible = "zmk,behavior-macro";
            #binding-cells = <0>; // Must be 0
            bindings = /* ... */;
        };

        // 1 param macro
        my_one_param_macro: my_one_param_macro {
            // ...
            compatible = "zmk,behavior-macro-one-param";
            #binding-cells = <1>; // Must be 1
            bindings = /* ... */;
        };

        // 2 params macro
        my_two_param_macro: my_two_param_macro {
            // ...
            compatible = "zmk,behavior-macro-two-param";
            #binding-cells = <2>; // Must be 2
            bindings = /* ... */;
        };
    };
};
```

### Parameters, Bindings and Controls

> There are special macro controls which must be used in order to forward received parameters to the macro's `bindings`. These controls are "one shot" and will determine how received parameters are used on the very next [non-macro control] behavior in the macro's `bindings` list.

Reviewer: ```suggestion
There are special macro controls which must be used in order to forward received parameters to the macro's `bindings`. These controls are "one shot" and will determine how received parameters are used on the very next (non-macro control) behavior in the macro's `bindings` list.
```

Author: Done

#### Layer + Modifier

```dts
/**
 * Temporarily switches to a layer (`&mo`) while a modifier is held.
 * Analogous to QMK's `LM()`, using a parameterized macro.
 *
 * Params:
 *  1. Layer to switch to
 *  2. Modifier to press while layer is active
 *
 * Example:
 *  `&lm NUM_LAYER LSHIFT`
 */
lm: lm {
    label = "LAYER_MOD";
    compatible = "zmk,behavior-macro-two-param";
    wait-ms = <0>;
    tap-ms = <0>;
    #binding-cells = <2>;
    bindings

>         = <&macro_param_1to1>,
>         <&macro_press &mo MACRO_PLACEHOLDER>,
>         <&macro_param_2to1>,
>         <&macro_press &kp MACRO_PLACEHOLDER>,
>         <&macro_pause_for_release>,
>         <&macro_param_2to1>,
>         <&macro_release &kp MACRO_PLACEHOLDER>,
>         <&macro_param_1to1>,
>         <&macro_release &mo MACRO_PLACEHOLDER>;

Reviewer: ```suggestion
        = <&macro_param_1to1>
        , <&macro_press &mo MACRO_PLACEHOLDER>
        , <&macro_param_2to1>
        , <&macro_press &kp MACRO_PLACEHOLDER>
        , <&macro_pause_for_release>
        , <&macro_param_2to1>
        , <&macro_release &kp MACRO_PLACEHOLDER>
        , <&macro_param_1to1>
        , <&macro_release &mo MACRO_PLACEHOLDER>
        ;
```
Style consistency with the rest of the macro docs

Author: Done
